sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -111,10 +116,18 @@ export class BatchingReceiver extends MessageReceiver {
         reject(error);
       };
 
+      let cleanupAbortSignal: (() => void) | undefined;
+
       // Final action to be performed after maxMessageCount is reached or the maxWaitTime is over.
       const finalAction = (this._finalActionHandler = (): void => {
         // clear finalActionHandler so that it can't be called multiple times.
         this._finalActionHandler = undefined;
+
+        if (cleanupAbortSignal) {
+          cleanupAbortSignal();
+          cleanupAbortSignal = undefined;
```

Comment:
Having the clean up being called from `finalAction()` ensures clean up on the happy path. What about cases when the promise returned from `receive()` is rejected?

Reply:
Great point, will push a fix, just waiting for the test run to complete. 

I'm just going to chain two promises together, rather than trying to intercept each exit from this particular promise.